src/grab_data.py

```diff
@@ -0,0 +1,239 @@
+        title_element = title_container.find_element_by_xpath('./h1/div')
+        title_text = title_element.get_attribute('textContent')
+        self.current_dict["title"] = title_text
+
+    def grab_all_users_and_counts(self) -> None:
+
+        ''' Defines a function that checks if a user is officially recognised
+            If official, runs official-user data grab, if not, runs non-official-user
+            data grab .
+        
+            Arguments: None
+            
+            Returns: None '''
+
+        if not (self.driver.find_elements_by_xpath('//div[@data-test-id="official-user-attribution"]')):
+            self.grab_non_official_user_and_count()
+        else:
+            self.grab_official_user_and_count()
+
+    def grab_official_user_and_count(self) -> None:
+
+        ''' Defines a function that grabs the poster name and follower count
+            from a Pinterest page if the user has official recognition
+            and adds them to the keys "poster_name" and "follower_count" in 
+            self.current_dict.
+            
+            Arguments: None
+            
+            Returns: None '''
+
+        container = self.driver.find_element_by_xpath('//div[@data-test-id="official-user-attribution"]')
+        poster_element = container.find_element_by_xpath('.//div[@class="tBJ dyH iFc yTZ pBj zDA IZT mWe CKL"]')
+        poster_name = poster_element.get_attribute('textContent')
+        self.current_dict["poster_name"] = poster_name
+        follower_element = container.find_element_by_xpath('.//div[@class="tBJ dyH iFc yTZ pBj zDA IZT swG"]')
+        followers = follower_element.get_attribute('textContent')
+        # If statement is needed as if there is no associated text I cannot use .split to grab only the value.
+        # Do not want the text "followers" on the end to clean the data somewhat.
+        if followers == '': # If the element has no associated text, there are no followers. Think this is redunant for official users.
+            self.current_dict["follower_count"] = 0
+        else:
+            follower_count = followers.split()[0]
+            self.current_dict["follower_count"] = follower_count
+
+    def grab_non_official_user_and_count(self) -> None:
+
+        ''' Defines a function that grabs the poster name and follower count
+            from a Pinterest page if the user doesnt have official recognition
+            and adds them to the keys "poster_name" and "follower_count" in 
+            self.current_dict.
+            
+            Arguments: None
+            
+            Returns: None '''
+
+        container = self.driver.find_element_by_xpath('//div[@data-test-id="CloseupUserRep"]//div[@data-test-id="user-rep"]')
+        poster_element = container.find_element_by_xpath('.//div[@class="tBJ dyH iFc yTZ pBj zDA IZT mWe"]')
+        poster_name = poster_element.get_attribute('textContent')
+        self.current_dict["poster_name"] = poster_name
```

Comment:
Can be put into a function with a dictionary of xpaths and dictionary keys passed to it to save to, same lines as 111/114.